From 8a3dfddbe2648606e929d7bfa001a07da53c1a4f Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@lsi.com>
Date: Wed, 14 May 2014 09:34:14 -0500
Subject: [PATCH 69/71] arch/arm/mach-axxia: Added the Redundant Parameters
 inthe Device Tree

git://git.yoctoproject.org/linux-yocto-3.10 standard/axxia/base
commit c5afddea9f3b725b0de617bc01eb0303a7570829

The partitioning of serial flash is what got changed in the device tree.
It now matches the images U-Boot expects (spl binary, parameters (two copies),
U-Boot environment (two copies), and U-Boot images.

This patch fixed the the difference in the Linux device tree for the
serial flash partitioning. It supports fall back images for the U-Boot
parameter file and U-Boot images, It is a new feature.

This change is required to support that feature.

Signed-off-by: John Jacques <john.jacques@lsi.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/boot/dts/axm5516-amarillo.dts |   16 ++++++++++++----
 1 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/axm5516-amarillo.dts b/arch/arm/boot/dts/axm5516-amarillo.dts
index aabc839..8d4ea09 100644
--- a/arch/arm/boot/dts/axm5516-amarillo.dts
+++ b/arch/arm/boot/dts/axm5516-amarillo.dts
@@ -100,19 +100,27 @@
 			reg = <0x0 0x40000>;
 		};
 		partition@40000 {
-			label = "parameters";
+			label = "parameters0";
 			reg = <0x40000 0x10000>;
 		};
+		partition@50000 {
+			label = "parameters1";
+			reg = <0x50000 0x10000>;
+		};
 		partition@60000 {
-			label = "env1";
+			label = "env0";
 			reg = <0x60000 0x10000>;
 		};
 		partition@70000 {
-			label = "env2";
+			label = "env1";
 			reg = <0x70000 0x10000>;
 		};
 		partition@100000 {
-			label = "u-boot";
+			label = "u-boot0";
+			reg = <0x100000 0x200000>;
+		};
+		partition@300000 {
+			label = "u-boot1";
 			reg = <0x100000 0x200000>;
 		};
 	};
-- 
1.7.5.4

